pkg/pipelinerunmetrics/metrics.go

```diff
@@ -256,40 +280,77 @@ func (r *Recorder) DurationAndCount(pr *v1beta1.PipelineRun, beforeCondition *ap
 	return nil
 }
 
-// RunningPipelineRuns logs the number of PipelineRuns running right now
+// RunningPipelineRuns logs the number of PipelineRuns running right now,
+// along with whether the number of PipelineRuns throttled by k8s scheduling.
 // returns an error if its failed to log the metrics
-func (r *Recorder) RunningPipelineRuns(lister listers.PipelineRunLister) error {
+func (r *Recorder) RunningPipelineRuns(prLister listers.PipelineRunLister, trLister listers.TaskRunLister) error {
 	r.mutex.Lock()
 	defer r.mutex.Unlock()
 
 	if !r.initialized {
 		return errors.New("ignoring the metrics recording, failed to initialize the metrics recorder")
 	}
 
-	prs, err := lister.List(labels.Everything())
+	prs, err := prLister.List(labels.Everything())
 	if err != nil {
 		return fmt.Errorf("failed to list pipelineruns while generating metrics : %w", err)
 	}
 
 	var runningPRs int
+	var prsThrottledByQuota int
+	var prsThrottledByNode int
 	for _, pr := range prs {
 		if !pr.IsDone() {
 			runningPRs++
 		}
+		if pr.IsDone() {
+			continue
+		}
+		for _, cr := range pr.Status.ChildReferences {
+			switch cr.Kind {
+			case pipeline.TaskRunControllerName:
+				trName := cr.Name
+				tr, trerr := trLister.TaskRuns(pr.Namespace).Get(trName)
+				if trerr == nil {
+					if tr.IsDone() || tr.IsCancelled() {
+						continue
+					}
+					succeedCondition := tr.Status.GetCondition(apis.ConditionSucceeded)
+					exitForLoop := false
+					if succeedCondition != nil && succeedCondition.Status == corev1.ConditionUnknown {
+						switch succeedCondition.Reason {
+						case pod.ReasonExceededResourceQuota:
+							prsThrottledByQuota++
+							// only count once per pipelinerun
+							exitForLoop = true
+						case pod.ReasonExceededNodeResources:
+							prsThrottledByNode++
+							// only count once per pipelinerun
+							exitForLoop = true
```

Comment:
Same here: Can we use break instead of exitForLoop?

Reply:
see https://github.com/tektoncd/pipeline/pull/6744/files#r1210620661